numpy/core/src/umath/umathmodule.c

```diff
@@ -164,39 +129,51 @@ ufunc_frompyfunc(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *NPY_UNUS
     if (i) {
         offset[0] += (sizeof(void *) - i);
     }
-    offset[1] = self->nargs;
-    i = (self->nargs % sizeof(void *));
+    offset[1] = nargs;
+    i = (nargs % sizeof(void *));
     if (i) {
         offset[1] += (sizeof(void *)-i);
     }
```

Comment:
This reuse of the variable `i` is pretty gross, as are the extra parens - but out of scope for this change I suppose.

Reply:
adopted

Reply:
We need the offsets for each piece of the allocated data, not just the overall size. Let's leave this for a future cleanup.